tests/tests_fabric/parity/test_parity_simple.py

```diff
@@ -118,10 +118,10 @@ def train_fabric(fabric):
     "precision, accelerator",
     [
         (32, "cpu"),
-        pytest.param(32, "cuda", marks=RunIf(min_cuda_gpus=1)),
+        pytest.param(32, "cuda", marks=RunIf(min_cuda_gpus=1, standalone=True)),
```

Comment:
these don't need to be marked as standalone. or do you see a good reason?
```suggestion
        pytest.param(32, "cuda", marks=RunIf(min_cuda_gpus=1)),
```

Reply:
@awaelchli I put a reason for it at the end of the PR post if you missed it. The tests fail because the `CUBLAS_WORKSPACE_CONFIG` needs to be set at the start of the python session.  I also included a small test to show this behavior in the expandable section.

Reply:
In that case It would be better to address this environment variable leak. Standalone tests are considerably slower to run so it's better to avoid them if possible

Reply:
I don't think it's really a leak, it happens because the variable isn't set at the beginning of the python run, and the error states it must be before any PyTorch applications.
```
RuntimeError: Deterministic behavior was enabled with either `torch.use_deterministic_algorithms(True)` or `at::Context::setDeterministicAlgorithms(true)`, but this operation is not deterministic because it uses CuBLAS and you have CUDA >= 10.2. To enable deterministic behavior in this case, you must set an environment variable before running your PyTorch application: CUBLAS_WORKSPACE_CONFIG=:4096:8 or CUBLAS_WORKSPACE_CONFIG=:16:8. For more information, go to https://docs.nvidia.com/cuda/cublas/index.html#cublasApi_reproducibility
``` 
I have reduced the reproducible error down to this
```python
import os
import torch 
import torch.nn as nn

def make_deterministic():
    os.environ["CUBLAS_WORKSPACE_CONFIG"] = ":4096:8"
    torch.use_deterministic_algorithms(True)
    torch.backends.cudnn.benchmark = False
    torch.manual_seed(1)
    torch.cuda.manual_seed(1)


layer = nn.Linear(32, 64).to("cuda")
data = torch.rand(32).to("cuda")
output = layer(data)

make_deterministic()  # setting deterministic algorithms after Torch operations causes error

output = layer(data)
```

Do you have suggestions for how I can solve this issue then? Thanks @awaelchli @carmocca

Reply:
Actually if I use `torch.use_deterministic_algorithms(True, warn_only=True)` its all okay and the parity test still passes so I guess this is a bug in Torch that it doesn't update?

Reply:
I think proper way to do it without setting the `warn_only=True` is to launch a subprocess

https://github.com/pytorch/pytorch/blob/e2923b521bd7fc67fa1e8f404397b20d1ec81668/torch/testing/_internal/common_utils.py#L1216-L1233
```
# Note that if a test requires CUDA to actually register the changed
# CUBLAS_WORKSPACE_CONFIG variable, a new subprocess must be created, because
# CUDA only checks the variable when the runtime initializes. Tests can be
# run inside a subprocess like so:
#
#   import subprocess, sys, os
#   script = '''
#   # Test code should go here
#   '''
#   try:
#       subprocess.check_output(
#           [sys.executable, '-c', script],
#           stderr=subprocess.STDOUT,
#           cwd=os.path.dirname(os.path.realpath(__file__)),
#           env=os.environ.copy())
#   except subprocess.CalledProcessError as e:
#       error_message = e.output.decode('utf-8')
#       # Handle exceptions raised by the subprocess here
```

Reply:
The current change in the latest commit LGTM too. 

> it happens because the variable isn't set at the beginning of the python run,

It would also be fine to move the make_deterministic() call to the beginning of `test_parity_single_device`.

Reply:
It would still have to be called twice anyways, because it resets the RNG seeds too?